### ISSUE TITLE
selectedListPeriod binding

### ISSUE THREAD

**Issue by Schnueggel** [doroppu/yoso-datepicker#7]
In line  454 it should be

```javascript
 scope.selectedPeriodList = scope.selectedPeriodList || [];
```

**Comment by lukas-zech-software** [comment]
**selectedPeriodList** is not meant to support two-way binding.
It is only to be read from the application.

If you want to have the datepicker have periods on initialisation, use **preselected-periods** 

This is by design, is that the application it self does not need to have the dependency on gfk-period to prefill the date-picker with periods. 

Therefore the **preselected-periods** use as simplified object interface to create the periods which can be fullfilled by pure Javascript without external dependencies.

´´´javascript
preselected-periods="[
      {periodMode:'d',start:'2015/01/1',end:'2015/02/1'},
      {periodMode:'m',start:'2015/01/1',end:'2015/03/1'}
  ]
´´´

If fully fledged two way binding supporting the gfk-periods is neccessary this can be built, by the cost of introducing the dependency to the app and also some parts of the code would need to be rewritten

**Comment by spanitz** [comment]
Without two-way binding you won't be able to watch on selectedPeriodList because the reference is destroyed by assigning [] to selectedPeriodList

**Comment by Schnueggel** [comment]
Just as a note assigning a new var to a scope variable will not destroy the watch but trigger it.  As least that would be what I expect

**Comment by lukas-zech-software** [comment]
implemented in 4.0.0
